### PR TITLE
{Error Improvement} Adjust error types

### PR DESCRIPTION
**Pull request by houk-ms**
**Description**  

This PR adjsut some error type names for error handling.

-  `UnclassifiedUserFault` refers to the fallback error type of UserFault. Developers could use this error type when they can not find other specific UserFault types for their error cases.
- `UnknonwError` refers to an error that we could not know which category(UserFault, ServiceError, ClientError) it belongs to. This will be only used in core's exception handler for error categorizing. 

**Testing Guide**  

Developer related. No changes in user experiences.

### CODE REVIEW

**Comment by houk-ms** [comment]
Need adjustments for the changes in #15776

**Review by fengzhou-msft — APPROVED** [review]

File: doc/error_handling_guidelines.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ The newly designed error types are provided in [azure/cli/core/azclierror.py](ht
 |    |    | -- FileOperationError
 |    |    | -- ManualInterrupt
 |    |    | -- ... More ...
-|    |    | -- UnknownError                     # fallback of UserFault related errors
+|    |    | -- UnclassifiedUserFault            # fallback of UserFault related errors
```
Comment (qwordy) [inline doc/error_handling_guidelines.md:39]:
```suggestion
|    |    | -- UnclassifiedError            # fallback of UserFault related errors
```

Comment (qwordy) [reply doc/error_handling_guidelines.md:39]: Others end with "Error"

**Comment by qwordy** [comment]
What's the difference between unclassified and unknown? The naming can be discussed.
E.g.
```
azure.core.exceptions.HttpResponseError: (InvalidParameter) The value of parameter type is invalid.
```
Will the error type be erased? I strongly suggest reserve the original error type. Otherwise, I don't know what's wrong given an unknown error.

**Review by haroldrandom — APPROVED** [review]

File: doc/error_handling_guidelines.md (hunk 1 above)
Comment (houk-ms) [reply doc/error_handling_guidelines.md:38]: It's ok, not such a rule to be end with `Error`

**Comment by houk-ms** [comment]
> What's the difference between unclassified and unknown? The naming can be discussed.
> E.g.
> 
> ```
> azure.core.exceptions.HttpResponseError: (InvalidParameter) The value of parameter type is invalid.
> ```
> 
> Will the error type be erased? I strongly suggest reserve the original error type. Otherwise, I don't know what's wrong given an unknown error.

Please see the PR description.

Yes, for PM's request, we may need remove the error type info. But the changes won't be applied in this release, so that we can have more time for discussion

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -17,3 +17,5 @@ This checklist is used to make sure that common guidelines for a pull request ar
 - [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).
 
 - [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).
+
+- [] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).
```
Comment (evelyn-ys) [inline .github/pull_request_template.md:21]:
```suggestion
- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).
```

File: src/azure-cli-core/azure/cli/core/util.py [hunk 3]
```diff
@@ -96,8 +96,8 @@ def handle_exception(ex):  # pylint: disable=too-many-locals, too-many-statement
             az_error = azclierror.ValidationError(error_msg)
 
         elif isinstance(ex, CLIError):
-            # TODO: Fine-grained analysis here for Unknown error
-            az_error = azclierror.UnknownError(error_msg)
+            # TODO: Fine-grained analysis here
+            az_error = azclierror.UnclassifiedUserFault(error_msg)
```
Comment (evelyn-ys) [inline src/azure-cli-core/azure/cli/core/util.py:100]: Why CLIError here is UserFault?

File: src/azure-cli-core/azure/cli/core/azclierror.py [hunk 4]
```diff
@@ -125,8 +133,8 @@ class InvalidArgumentValueError(UserFault):
     pass
 
 
-class ArgumentParseError(UserFault):
-    """ Fallback of the argument parsing related errors.
+class ArgumentUsageError(UserFault):
```
Comment (yungezz) [inline src/azure-cli-core/azure/cli/core/azclierror.py:136]: ArgumentParseError seems making more sense.

Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/azclierror.py:136]: Want use this type to cover the original UsageError, which usually are the across arguement error

File: src/azure-cli-core/azure/cli/core/util.py (hunk 3 above)
Comment (houk-ms) [reply src/azure-cli-core/azure/cli/core/util.py:100]: Ideally, `CLIError` will not show up here any more. Temporarily count the CLIError here as UserFaults for better Telemetry analysis.

**Review by Juliehzl — APPROVED** [review]
LGTM

**Review by jiasli — APPROVED** [review]

**Comment by qwordy** [comment]
> 
> 
> > What's the difference between unclassified and unknown? The naming can be discussed.
> > E.g.
> > ```
> > azure.core.exceptions.HttpResponseError: (InvalidParameter) The value of parameter type is invalid.
> > ```
> > 
> > 
> > Will the error type be erased? I strongly suggest reserve the original error type. Otherwise, I don't know what's wrong given an unknown error.
> 
> Please see the PR description for their difference.
> 
> Yes, for PM's request, we may need remove the error type info. But the changes won't be applied in this release, so that we can have more time for discussion

Error type is important. If it is removed, I can't help users resolve their issues. At least, we need to make sure error type and stack trace are available in --debug logs. @Our PM

**Pull request merged**